File: cpp/k.cpp
#include<iostream>
using namespace std;


int main(){
  int n;
  int t0=0,x0=0,y0=0;

  bool ans=true;
  cin>>n;
  for(int i=0;i<n;i++){
    int t,x,y;
    cin>>t>>x>>y;
    if((t-t0)<abs(x-x0)+abs(y-y0)||(t-t0)%2!=abs(x+y-x0-y0)%2)
      ans = false;
    t0=t;
    x0=x;
    y0=y;
  }

  if(ans)
    cout<<"Yes"<<endl;
  else
    cout<<"No"<<endl;

}

File: atcoder/ABS/php/k.php
<?php

fscanf(STDIN,"%d",$n);

list($time,$a,$b) =[0,0,0];


while($n--){
  list($t,$x,$y) = explode(" ",fgets(STDIN));

  $d = abs($a-$x)+abs($b-$y);

  if(!($t-$time>=$d&&($t-$time)%2==$d%2)){
    print"No\n";
    return;
  }
  $time=$t;
  $a=$x;
  $b=$y;
}

print "Yes\n";

?>

